common/test/acceptance/pages/studio/settings_certificates.py

```diff
@@ -171,37 +172,75 @@ def click_confirmation_prompt_primary_button(self):
         self.wait_for_ajax()
 
 
-class Certificate(object):
+class CertificateSectionPage(CertificatesPage):
     """
-    Certificate page object wrapper
+    CertificateSectionPage is the certificate section within Certificates page, There might be multiple certificates
+    in a Certificates Page so this section object can be used to used to identify unique certificate and apply
+    operations on it.
     """
 
-    def __init__(self, page, prefix, index):
-        self.page = page
+    def __init__(self, container_certificates_page, prefix, index):
+        """
+        Initialize CertificateSection Page
+
+        :param container_certificates_page: Container Page Object of the certificate section
+        :param prefix: css selector of the container element
+        :param index: index of section in the certificate list on the page
+
+        :return:
+        """
         self.selector = prefix + ' .certificates-list-item-{}'.format(index)
         self.index = index
+        self.container = container_certificates_page
+
+        super(CertificateSectionPage, self).__init__(
+            container_certificates_page.browser,
+            **container_certificates_page.course_info
+        )
+
+    def is_browser_on_page(self):
```

Comment:
It's actually not a good pattern to have a wait in an `is_browser_on_page` method. This should just be a boolean, and any waiting would happen automatically when you do `my_page.visit()` or you would explicitly run `wait_for_page_to_load()` in a given test.